### PR TITLE
Fixed nullable internal link and added serverside validation

### PR DESCRIPTION
**Pull request by wachterjohannes**
__tasks:__

- [x] test coverage
- [ ] gather feedback for my changes

__informations:__

| q                | a
| ---------------- | ---
| Fixed tickets    | none
| BC breaks        | none
| Documentation PR | none

### CODE REVIEW

File: src/Sulu/Bundle/ContentBundle/Resources/public/js/components/content/settings/main.js [hunk 1]
```diff
@@ -360,12 +372,66 @@ define(['app-config', 'sulusecurity/components/users/models/user'], function(App
...
+                return this.validateInternal(data);
+            } else if (data.nodeType === TYPE_EXTERNAL) {
+                return this.validateExternal(data);
+            }
+
+            return true;
+        },
+
+        validateInternal: function(data) {
+            var result = true;
```
Comment (danrot) [inline src/Sulu/Bundle/ContentBundle/Resources/public/js/components/content/settings/main.js]: Minor code style issue: I would not introduce this variable:

```javascript
            if (!data.title) {
                this.sandbox.dom.addClass(constants.internalLink.titleContainer, constants.validateErrorClass);
                return false;
            }

            if (!data.internal_link) {
                this.sandbox.dom.addClass(constants.internalLink.linkContainer, constants.validateErrorClass);
                return false;
            }

            return true;
```

Should do the same job, with less lines, memory and in less time :wink:

File: src/Sulu/Bundle/ContentBundle/Resources/public/js/components/content/settings/main.js [hunk 2]
```diff
@@ -360,12 +372,66 @@ define(['app-config', 'sulusecurity/components/users/models/user'], function(App
...
+                result = false;
+                this.sandbox.dom.addClass(constants.externalLink.titleContainer, constants.validateErrorClass);
+            }
+
+            if (!data.external) {
+                result = false;
+                this.sandbox.dom.addClass(constants.externalLink.linkContainer, constants.validateErrorClass);
+            }
+
+            return result;
```
Comment (danrot) [inline src/Sulu/Bundle/ContentBundle/Resources/public/js/components/content/settings/main.js]: Same as above

File: src/Sulu/Bundle/ContentBundle/Resources/public/js/components/content/settings/main.js (hunk 1 above)
Comment (wachterjohannes) [inline src/Sulu/Bundle/ContentBundle/Resources/public/js/components/content/settings/main.js]: if both is empty, both should be red ... so only my version works for this usecase!

File: src/Sulu/Bundle/ContentBundle/Resources/public/js/components/content/settings/main.js (hunk 2 above)
Comment (wachterjohannes) [inline src/Sulu/Bundle/ContentBundle/Resources/public/js/components/content/settings/main.js]: same as above (:

File: tests/Sulu/Component/Content/Document/Subscriber/RedirectTypeSubscriberTest.php [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+        $node->setProperty('i18n:en-nodeType', RedirectType::INTERNAL, PropertyType::LONG)->shouldBeCalled();
+        $node->setProperty('i18n:en-external', null)->shouldBeCalled();
+        $node->setProperty('i18n:en-internal_link', $redirectNode->reveal())->shouldBeCalled();
+        $node->getPropertyValueWithDefault('i18n:en-nodeType', 1)->willReturn(RedirectType::INTERNAL);
+        $node->getPropertyValueWithDefault('i18n:en-internal_link', null)->willReturn($redirectNode->reveal());
+
+        $subscriber->doPersist($event->reveal());
+    }
+
+    public function testPersistWithTypeNoneMissingLink()
```
Comment (danrot) [inline tests/Sulu/Component/Content/Document/Subscriber/RedirectTypeSubscriberTest.php]: I guess that should be `testPersistWithtypeInternalMissingLink`

Comment (wachterjohannes) [inline tests/Sulu/Component/Content/Document/Subscriber/RedirectTypeSubscriberTest.php]: yes (:

**Pull request merged**